File: Backend/EComCore.Domain/Entities/Review.cs
namespace EComCore.Domain.Entities;

public class Review
{
    public int Id { get; set; }
    public int ProductId { get; set; }
    public Product Product { get; set; }
    public int MemberId { get; set; }
    public User User { get; set; }
    public int Rating { get; set; }
    public string Comment { get; set; }
    public string Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public bool IsDeleted { get; set; }

}

File: client/src/types/review.ts
export interface Review {
  id: number;
  productId: number;
  productName: string;
  userId: number;
  userName: string;
  rating: number;
  comment: string;
  status: string;
  moderationReason: string;
  createdAt: Date;
  updatedAt: Date;
}
export interface CreateReview {
  productId: number;
  userId: number;
  comment: string;
  rating: number;
}

export interface UpdateReview {
  id: number;
  rating?: number;
  comment?: string;
  status?: ReviewStatus;
  moderationReason?: string;
}

export enum ReviewStatus {
  Pending = "Pending",
  Approved = "Approved",
  Rejected = "Rejected",
}
